shared/AppInsightsCommon/src/Offline.ts

```diff
@@ -1,11 +1,40 @@
 import {
-    createUniqueNamespace, eventOff, eventOn, getDocument, getNavigator, getWindow, isNullOrUndefined, isUndefined, mergeEvtNamespace
+    IUnloadHook, createUniqueNamespace, eventOff, eventOn, getDocument, getNavigator, getWindow, isNullOrUndefined, isUndefined,
+    mergeEvtNamespace
 } from "@microsoft/applicationinsights-core-js";
 
+/**
+ * this is the callback that will be called when the network status changes
+ * @param onlineState this is the current network running state
+ */
+export type OfflineCallback = (onlineState: IOfflineState) => void;
+
+/**
+ * This is the enum for the different network states current ai is experiencing
+ */
+export const enum eOfflineValue {
+    Unknown = 0,
+    Online = 1,
+    Offline = 2
+}
+/**
+ * This is the interface for the Offline state
+ * runtime state is retrieved from the browser state
+ * user state is set by the user
+*/
+export interface IOfflineState {
+    readonly isOnline: boolean;
+    readonly rState: eOfflineValue; // runtime state
+    readonly uState: eOfflineValue; // user state
+}
+
 export interface IOfflineListener {
     isOnline: () => boolean;
     isListening: () => boolean;
     unload: () => void;
+    addListener(callback: OfflineCallback): IUnloadHook;
+    userSetOnlineState: (uState: eOfflineValue) => void;
```

Comment:
Lets call this `setOnlineState` as we are not letting them set the runtime state.